Approximate_solvers.ipynb

```diff
@@ -36,6 +36,51 @@
+    "\n",
+    "\\begin{align}\\label{FVupdate}\n",
+    "Q_j^{n+1} = Q_j^n - \\frac{\\Delta t_n}{\\Delta x_j} (F_{j+1/2}^n - F_{j-1/2}^n),\n",
+    "\\end{align}\n",
+    "\n",
+    "for some definition of the *numerical flux* $F_{j-1/2}^n$, typically based on $Q_{j-1}^n$ and $Q_j^n$ (and possibly other nearby cell values at time $t_n$).\n",
+    "\n",
+    "Dividing (\\ref{FVupdate}) by $\\Delta t_n$ and rearrraning, this form can be viewed as a discretization of $q_t + f(q)_x = 0$, provided the numerical flux is *consistent* with the true flux $f(q)$ in a suitable manner. In particular if the $Q_i$ used in defining $F_{j-1/2}^n$ are all equal to the same value $\\bar q$, then $F_{j-1/2}^n$ should reduce to $f(\\bar q)$.\n",
+    "\n",
+    "A big advantage of using conservation form is that the numerical method is conservative. The sum $\\sum \\Delta x_j Q_j^n$ approximates the integral $\\int q(x,t_n)\\,dx$.  Multiplying (\\ref{FVupdate}) by $\\Delta x_j$ and summing shows that at time $t_{n+1}$ this sum only changes due to fluxes at the boundaries of the region in question (due to cancellation of the flux differences when summing), a property shared with the true solution.  For problems with shock waves, using methods in conservation form is particularly important since nonconservative formulations can lead to methods that converge to discontinuous solutions that look fine but are not correct, e.g. the shock wave might propagate at entirely the wrong speed. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Godunov's method\n",
+    "\n",
+    "Trying to compute numerical approximations to hyperbolic problems with strong shock waves is challenging because of the discontinuities in the solution --- classical interpretations of $(F_{j+1/2}^n - F_{j-1/2}^n)/\\Delta x_j \\approx \\partial f/ \\partial x$ break down, oscillations near discontinuities often appear, and methods can easily go catastrophically unstable.\n",
+    "\n",
+    "The landmark paper of Godunov <cite data-cite=\"godunov\"><a href=\"riemann.html#godunov\">(Godunov 1959)</a></cite> was the first to suggest using the solution to Riemann problems in defining the numerical flux: $F_{j-1/2}^n$ is obtained by evaluationg $f(Q_{j-1/2}^*)$, where $Q_{j-1/2}^*$ is the Riemann solution evaluated along the ray $x/t = 0$ after the Riemann problem is solved between states $q_\\ell=Q_{j-1}^n$ and $q_r=Q_j^n$. If the numerical solution is defined as a piecewise constant function with value $Q_j^n$ in the $j$th cell, then the exact solution takes this value along the cell interface $x_{j-1/2}$ for sufficiently small later times $t > t_n$ (until waves from other cell interfaces begin to interact).\n",
```

Comment:
evaluating

Reply:
It seems necessary here to clarify that the Riemann problem is solved in a coordinate system where $x_{j-1/2}=0$.

Reply:
I tried to word this part as solving the standard Riemann problem (as discussed up to now) with states $q_\ell$ and $q_r$, and then the the state along $x/t = 0$ in this solution is the one that appears along the cell edge on the finite volume grid.  I tried rewording this a bit, see if it is any clearer now.